cicecore/cicedynB/dynamics/ice_dyn_evp.F90

```diff
@@ -219,25 +295,58 @@ subroutine evp (dt)
       ! convert fields from T to U grid
       !-----------------------------------------------------------------
 
-      call to_ugrid(tmass,umass)
-      call to_ugrid(aice_init, aiu)
-
+      call grid_average_X2Y('F',tmass,'T',umass,'U')
+      call grid_average_X2Y('F',aice_init,'T',aiu,'U')
+      call grid_average_X2Y('S',uocn,grid_ocn_dynu,uocnU,'U')
+      call grid_average_X2Y('S',vocn,grid_ocn_dynv,vocnU,'U')
+      call grid_average_X2Y('S',ss_tltx,grid_ocn_dynu,ss_tltxU,'U')
+      call grid_average_X2Y('S',ss_tlty,grid_ocn_dynv,ss_tltyU,'U')
```

Comment:
You still need to do the copy into the output variables in that case, i.e. uocnU=uocn.  This is done automatically in grid_average_X2Y.  Otherwise, you have to do some complex switching of variable name to use uocnU if grid_ocn_dynu/=U and uocn otherwise.  This copy is cheap and is outside subcycling.  Performance improvements will come inside the subcycling.